borgmatic/tests/unit/config/test_validate.py

```diff
@@ -23,13 +23,37 @@ def test_apply_logical_validation_raises_if_archive_name_format_present_without_
             },
         )
 
+def test_apply_logical_validation_raises_if_archive_name_format_present_without_retention_prefix():
```

Comment:
This is going to be a breaking change for existing configurations. Users who previously used `archive_name_format` will need to update their config to add the new `consistency.archive_prefix` option.

How do you handle configuration schema compatibility?

Reply:
I generally don't! I've only one had one majorly breaking change, and that was the transition from an INI-like config file to YAML. And for that I included (and required using) a transition tool.

This is more minor, so it might be okay to just ship it and include a decent error message. See my separate comment though about whether this validation is even necessary.

Reply:
I've taken the warn approach. I've modified the logging configuration to default to "Warn" level, instead of error. 

I think this should give enough visibility to the change.